src/app/src/components/DrawTools/DrawTools.js

```diff
@@ -9,7 +12,27 @@ import useEditingPolygon from './useEditingPolygon';
 import useGeocoderResult from './useGeocoderResult';
 import useTrackMapZoom from './useTrackMapZoom';
 
-export default function DrawTools() {
+import { setPolygon } from '../../store/mapSlice';
+
+export default function DrawTools({ details }) {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
```

Comment:
Consider adding a comment block here summarizing what this `useEffect` does, which will be handy for future reviews.